### PR TITLE
docs: update README.md, fix example code.

### PR DESCRIPTION
**Pull request by laoer536**

### CODE REVIEW

**Comment by antfu** [comment]
What has been fixed? Can you explain a bit?

File: README.md [hunk 1]
```diff
@@ -53,8 +53,8 @@ const { config, sources } = await loadConfig({
     {
       files: 'vite.config',
       async rewrite(config) {
-        const config = await (typeof config === 'function' ? config() : config)
```
Comment (laoer536) [inline README.md:56]: Rewrite function has a 'config' parameter, but the next line use 'const' declare again.

**Comment by laoer536** [comment]
> What has been fixed? Can you explain a bit?

The 'rewrite' function has a 'config' parameter, but the next line use 'const' declare again.

File: README.md [hunk 2]
```diff
@@ -53,8 +53,8 @@ const { config, sources } = await loadConfig({
...
-        return config?.my
+        const configRes = typeof config === 'function' ? await config() : config
+        return configRes?.my
```
Comment (antfu) [inline README.md:57]:
```suggestion
        const resolved = await (typeof config === 'function' ? config() : config)
        return resolved?.my
```

**Pull request merged**